ray-operator/test/support/ray.go

```diff
@@ -247,3 +249,47 @@ func Gateway(t Test, namespace, name string) func() (*gwv1.Gateway, error) {
 		return GetGateway(t, namespace, name)
 	}
 }
+
+// VerifyContainerAuthTokenEnvVars verifies that the specified container has the correct auth token environment variables.
+// This is a common helper function used across all auth-related E2E tests.
+func VerifyContainerAuthTokenEnvVars(t Test, rayCluster *rayv1.RayCluster, pod *corev1.Pod, containerIndex int) {
+	t.T().Helper()
+	g := NewWithT(t.T())
+
+	g.Expect(len(pod.Spec.Containers)).To(BeNumerically(">", containerIndex),
```

Comment:
I think we can remove this assertion and the`containerIndex` parameter, and instead just pass in `corev1.Container`.